packages/ui/src/SelectField.tsx

```diff
@@ -106,28 +106,45 @@ export type SelectFieldCoreDynamicProps<V> =
   | {
       isClearable: true
       isMulti?: false
-      value: SelectFieldOption<V> | null
-      onChange: (newValue: SelectFieldOption<V> | null) => void
+      value: V | null
+      onChange: (newValue: V | null) => void
     }
   | {
       isClearable?: false
       isMulti?: false
-      value: SelectFieldOption<V>
-      onChange: (newValue: SelectFieldOption<V>) => void
+      value: V
+      onChange: (newValue: V) => void
     }
   | {
       isClearable: true
       isMulti: true
-      value: SelectFieldOption<V>[] | null
-      onChange: (newValue: SelectFieldOption<V>[] | null) => void
+      value: V[] | null
+      onChange: (newValue: V[] | null) => void
```

Comment:
Since empty `multi select` now returns an empty array when there's nothing selected, we might want to remove all these ugly `null`'s from other places:

```suggestion
      value: V[]
      onChange: (newValue: V[]) => void
```